src/compiler/checker.ts

```diff
@@ -17757,11 +17775,20 @@ namespace ts {
                 // However, resolveNameHelper will continue and call this callback again, so we'll eventually get a correct suggestion.
                 return symbol || getSpellingSuggestionForName(unescapeLeadingUnderscores(name), arrayFrom(symbols.values()), meaning);
             });
-            return result && symbolName(result);
+            return result;
+        }
+
+        function getSuggestionForNonexistentSymbol(location: Node | undefined, outerName: __String, meaning: SymbolFlags): string | undefined {

```

Comment:
we should have made these all internal :(

Reply:
Yeah. 😡

We still can?

Reply:
it is a breaking change..

Reply:
but yes we can